lib/ui/window/pointer_data_packet.cc

```diff
@@ -20,4 +20,14 @@ void PointerDataPacket::SetPointerData(size_t i, const PointerData& data) {
   memcpy(&data_[i * sizeof(PointerData)], &data, sizeof(PointerData));
 }
 
+PointerData PointerDataPacket::GetPointerData(size_t i) {
+  PointerData result;
+  memcpy(&result, &data_[i * sizeof(PointerData)], sizeof(PointerData));
```

Comment:
When I read "eliminate unsafe memcpy", I expected some form of bounds checking. If `i` is too large, you'll still end up reading junk data. Not sure how this is any safer. Perhaps add the bounds checks?

Reply:
While you're at it, perhaps also add bounds checks to the setter?

Reply:
I want that initially, but the old code did not do any such checks (due to performance? or other reasons?) If you decide to add it, I will definitely add.

Reply:
checks added